lib/garbage_collector.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'bintray_client'
+require_relative 'github_client'
+
+class GarbageCollector
+
+  def initialize(bintray_user, bintray_key, verbose = false, dry_run = false)
+    @bintray_user = bintray_user
+    @bintray_key = bintray_key
+    @verbose = verbose
+    @dry_run = dry_run
+
+    @bintray = BintrayClient.new(bintray_user, bintray_key)
+    @github = GitHubClient.new
+  end
+
+  def run(projects)
+    projects.each do |project|
+      log "Looking for versions to delete for #{project[:bintray]}"
+
+      dev_versions = @bintray.get_bintray_versions(project[:bintray])
+
+      if dev_versions.empty?
+        log "No versions found. Moving on."
+        next
+      end
+
+      log "Found versions for PR(s): #{dev_versions.keys.join(', ')}"
+
+      dev_versions.each do |pr_id, versions|
+        log "Checking if versions for PR #{pr_id} should be deleted"
+
+        if @github.is_pr_closed?(project[:github], pr_id)
+          log "Deleting versions for PR #{pr_id}"
+        else
+          log "PR #{pr_id} is not closed. Skipping."
```

Comment:
nit 😛 
```suggestion
          log "PR #{pr_id} is still open. Skipping."
```

Reply:
PS: Also super extra nitpick, but could be nice having a simple visual way (some prefix char? or use ANSI colors? 😛 ) to easily differentiate at first glance log messages that are announces ("Checking xxx..." – adding "..." to indicate it might take the time of an HTTP request to do it) vs the ones that are actions/replies.

For example:
```
Looking for versions to delete for *******/maven/utils-experiment...
> Found versions for PR(s): 3, 2, 40
Checking if versions for PR 3 should be deleted...
> PR 3 is not closed. Skipping.
Checking if versions for PR 2 should be deleted...
> Deleting versions for PR 2...
> Deleted version 2-d869dc6
Checking if versions for PR 40 should be deleted...
> PR 40 is not closed. Skipping.
```

_That's my OCD speaking so feel free to ignore_ 😅

Reply:
Thanks for the suggestions @Automattic/jetpack-design 

- Active positive language FTW 👍 
- I like the prefix char idea (I love colors, too, but might be a bit of overhead for this tool, at least right now) I thought about using `\t` originally, but left it out because I didn't like the look of it. This is much better!